OpenInActionExtension/Action.js

```diff
@@ -0,0 +1,47 @@
+//
+//  Action.js
+//  OpenInActionExtension
+//
+//  Created by Marcus Kida on 03.01.23.
+//
+
+var Action = function() {};
+
+Action.prototype = {
+    
+    run: function(arguments) {
+        var payload = {
+            "username": detectUsername(),
+            "url": document.documentURI
+        }
+
+        arguments.completionFunction(payload)
+    },
+    
+    finalize: function(arguments) {
+        let alertMessage = arguments["alert"]
+        if (alertMessage) {
+            alert(alertMessage)
+        } else {
+            window.location = arguments["openURL"]
+        }
+    }
+    
+};
+
+function detectUsername() {
+    var uriUsername = document.documentURI.match("@(.+)@([a-z0-9]+\.[a-z0-9]+)")
```

Comment:
More of a question: Is this like the _real_ regex to get Mastodon-usernames? I'm asking, as something like `@mailto:email😊@mastodon.social@mastodon.social` (or worse, like crashing strings) would be valid, too, I guess?

I'm talking about the `(.+)`. But to be honest, I have no idea, which characters a valid, I was expecting something like `[a-zA-Z0-9.]` here.

[Special characters in usernames](https://github.com/mastodon/mastodon/issues/6830) seem to be a hot topic

Reply:
I've picked this deliberately because I wasn't sure if a username can't contain anything else than ascii chars, but that said, maybe it's a good idea to double check and this is what we're using in the iOS app:

`"(?:@([a-zA-Z0-9_]+)(@[a-zA-Z0-9_.-]+)?|#([^\\s.]+))"`

So I'll repurpose the same in the JS code as well.